components/brave_wallet/browser/brave_wallet_p3a.cc

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "base/metrics/histogram_macros.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_p3a.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+#include "brave/components/brave_wallet/browser/keyring_service.h"
+#include "brave/components/brave_wallet/browser/pref_names.h"
+#include "components/prefs/pref_service.h"
+
+namespace brave_wallet {
+
+// Has the Wallet keyring been created?
+// i) Yes, ii) No
+void RecordKeyringCreated(mojom::KeyringInfoPtr keyring_info) {
+  const int max_bucket = 2;
+  UMA_HISTOGRAM_EXACT_LINEAR(
+      "Brave.Wallet.KeyringCreated",
+      static_cast<int>(keyring_info->is_default_keyring_created), max_bucket);
+}
+
+// What is the DefaultWalletSetting?
+// i) AskDeprecated, ii) None, ii) CryptoWallets,
+// iv) BraveWalletPreferExtension, v) BraveWallet
+void RecordDefaultWalletSetting(PrefService* pref_service) {
+  const int max_bucket = 5;
```

Comment:
Maybe `static_assert(max_bucket == brave_wallet::mojom::DefaultWallet::kMaxValue)`? Mostly to document where these values are defined.

Reply:
Good tip! I'll add the assertion

Reply:
That would be fine too. It looks like the mojom codegen adds a kMaxValue automatically. `src/out/Component/gen/brave/components/brave_wallet/common/brave_wallet.mojom-shared.h` has
```c++
enum class DefaultWallet : int32_t {
  AskDeprecated = 0,
  // more values here...
  BraveWallet = 4,
  kMinValue = 0,
  kMaxValue = 4,
};
```